### ISSUE TITLE
Unable to install STS4.7.0 with Eclipse 2020-06

### ISSUE THREAD

**Issue by chattling** [spring-projects/sts4#495]
I used freshly installed Eclipse 2020-06 for Java developers. No other plugins installed. Opened with new workspace.
Installed Spring Tools 4 (aka Spring Tool Suite 4) 4.7.0.RELEASE from Eclipse Marketplace accessible via Help menu.
The installation went well. At the end of installation it asks to restart to apply changes.
After "Restart Now" existing Eclipse closed and splash screen of new Eclipse displayed. However the loading does not complete. It always (retried opening Eclipse multiple times) ending up in error message:

![image](https://user-images.githubusercontent.com/34875079/85859027-328db280-b782-11ea-8004-f5f82ac593fa.png)

Following the log file in the message, full log is in attached.
[log.txt](https://github.com/spring-projects/sts4/files/4837036/log.txt)

**Comment by kdvolder** [comment]
Can you try with the full distribution builld of STS instead of installing it into Eclipse from update site? It would be interesting to try this to know if that works better.

You can find the current / latest distribution here: https://download.springsource.com/release/STS4/4.7.0.RELEASE/dist/e4.16/spring-tool-suite-4-4.7.0.RELEASE-e4.16.0-win32.win32.x86_64.self-extracting.jar

**Comment by chattling** [comment]
Full distribution build from link that you provided opens with no issues.
I prefer to install it from Marketplace/update site though.

**Comment by kdvolder** [comment]
> I prefer to install it from Marketplace/update site though.

I understand. And of course that should work too and we will need to try to understand why it didn't work for you. But at least in the mean time you can get back to work using the distribution build.

**Comment by chattling** [comment]
> I understand. And of course that should work too and we will need to try to understand why it didn't work for you. But at least in the mean time you can get back to work using the distribution build.

I'm sticking for now with Eclipse 2020-03. If you need me to provide any details to troubleshoot update site version with Eclipse 2020-06, please don't hesitate to ask.

**Comment by eliseucartaxo** [comment]
I'm with the same problem.

**Comment by mauromol** [comment]
I don't know whether the problems are related, if not please tell me and I'll open a new bug report.
Trying to update from STS 4.6.1 to STS 4.7.0 in an Eclipse 2019-12 install fails with:

```
Cannot complete the install because one or more required items could not be found.
  Software being installed: Spring Tool Suite 4 Main Feature 4.7.0.202006181158-RELEASE (org.springframework.boot.ide.main.feature.feature.group 4.7.0.202006181158-RELEASE)
  Missing requirement: jersey-media-json-jackson 2.30.1.v20200513-1859 (org.glassfish.jersey.media.jersey-media-json-jackson 2.30.1.v20200513-1859) requires 'java.package; com.fasterxml.jackson.core [2.10.0,3.0.0)' but it could not be found
  Cannot satisfy dependency:
    From: Spring Tool Suite 4 Main Feature 4.7.0.202006181158-RELEASE (org.springframework.boot.ide.main.feature.feature.group 4.7.0.202006181158-RELEASE)
    To: org.eclipse.equinox.p2.iu; org.springframework.ide.eclipse.boot.wizard [3.9.13.202006180736-RELEASE,3.9.13.202006180736-RELEASE]
  Cannot satisfy dependency:
    From: Boot UI Support 3.9.13.202006180736-RELEASE (org.springframework.ide.eclipse.boot.wizard 3.9.13.202006180736-RELEASE)
    To: java.package; org.glassfish.jersey.jackson 2.30.0
```

However, my update site list seems to be ok and contains the following enabled sites:

Name | Location
------- | -----------
2019-12 | http://download.eclipse.org/releases/2019-12
CodeTogether CI Update Site | https://www.codetogether.com/updates/ci/
DevStyle CI Update Site | http://www.genuitec.com/updates/devstyle/ci/
org.codehaus.groovy.eclipse.site | https://dist.springsource.org/snapshot/GRECLIPSE/e4.14
org.zeroturnaround.eclipse.updatesite | http://update.zeroturnaround.com/update-site/
Spring Tool Suite 4 | https://download.springsource.com/release/TOOLS/sts4/update/e4.14/
Subclipse Latest Release | https://dl.bintray.com/subclipse/releases/subclipse/latest/
The Eclipse Project Updates | http://download.eclipse.org/eclipse/updates/4.14
update site: http://regex-util.sourceforge.net/update/ | http://regex-util.sourceforge.net/update/

**Comment by kdvolder** [comment]
@mauromol As possible workaround for you, maybe try installing the missing jersey client dependency from eclipse orbit. 
https://download.eclipse.org/tools/orbit/downloads/drops/R20200529191137/

Caution, this version of jersey client is rather newish and *only* available in the latest orbit update site. The reason you are not finding it is probably because it was not part of Eclipse update site ecosystem prior to Eclipse 4.16. There is some risk in trying to install bundles from the latest orbits site into an older Eclipse as you are implicitly setting up untested combinations of older and newer dependencies. So exercise some caution... I would *only* install just enough (jersey-client) to satisfy the dependency and then delete the orbits site again to avoid pulling in more things accidentally.

I'm not sure why your set of update sites doesn't already provide the jersey client dependency though, I *thought* it should have been included into STS 4 update site for e 4.14.If you feel 'adventurous' perhaps you might want to try an Eclipse 4.16 based setup instead. (Though I understand you probably made a conscious decision to stay on the older Eclipse, the choice is yours of course :-).

**Comment by kdvolder** [comment]
I tried the steps @chattling describes and the problem is reproducible for me also on Linux.

**Comment by chattling** [comment]
Another observation. I took my Eclipse 2020-03 with STS 4.6.0 installed. Then I used Eclipse Help -> Check for Updates menu to update itself. Eclipse got upgraded to 2020-06 (STS is still 4.6.0).
Then I opened Eclipse Marketplace and it gave me an option to upgrade STS to 4.7.0. I went thru its upgrade screens.
Now I have Eclipse 2020-06 with STS 4.7.0 working.
So via upgrade path 2020-06 and STS 4.7.0 coexist. Not as new installation though.

**Comment by kdvolder** [comment]
I filed a bug here: https://bugs.eclipse.org/bugs/show_bug.cgi?id=564803

This seems to me like it is most likely a bug in the Eclipse platform or perhaps osgi/equinox update site mechanics. TBH, I'm not really sure.

Doing as @chattling said exactly results in a broken Eclipse that cannot startup anymore. Though it doesn't actually seem that the specific combination of plugins is what breaks it because when I add `-clean` option as the first line to my `STS.ini` file. Then STS starts up fine again. We can then remove the `-clean` option again and everything then continues to work fine on successive restarts. The clean option basically tells Eclipse clear out cached state from OSGI and figure out the dependencies of all the bundles from scratch. It doesn't actually change the bundles.

Anyhow...

**Simple workaround is to add `-clean` argument as first line to `STS.ini.**

**Comment by kdvolder** [comment]
Since I beleave this problem isn't a bug in STS itself. I'm closing this ticket. Please direct further comments/information to Eclipse bugzilla. 

@mauromol Your problem seems unrelated. It may be a packaging issue with STS (something missing from our update site). I will create a new github ticket for this and investigate.

**Comment by mauromol** [comment]
> I'm not sure why your set of update sites doesn't already provide the jersey client dependency though, I _thought_ it should have been included into STS 4 update site for e 4.14.If you feel 'adventurous' perhaps you might want to try an Eclipse 4.16 based setup instead. (Though I understand you probably made a conscious decision to stay on the older Eclipse, the choice is yours of course :-).

Hi Kris, usually when I upgrade to a newer Eclipse version I still keep an old one instance and migrate my workbenches incrementally, until I've determined that the new version is safe to use. This is why I'm still trying to update my 2019-12 instance while testing 2020-06. Since "check for updates" proposes STS 4.7.0 I believe it is supported for the old Eclipse version too, this is why I'm here.
I'll follow up on #497, thanks.

BTW: FWIW I could install STS 4.7.0 on Eclipse 2020-06 with no issue at all. I didn't use the Eclipse Marketplace, though, I went with the update site. My base was not one of the packages available for download at https://www.eclipse.org/downloads/, but rather an Eclipse Runtime Platform base on top of which I had already installed many other plugins before installing STS 4.7.0. Just as a note.
